org.lflang/src/org/lflang/FileConfig.java

```diff
@@ -192,6 +193,28 @@ public FileConfig(Resource resource, IFileSystemAccess2 fsa, IGeneratorContext c
         this.iResource = getIResource(resource);
     }
     
+    protected FileConfig(FileConfig fileConfig) throws IOException {
```

Comment:
Document that this is a copy constructor?

Reply:
Why is a copy constructor needed? In my view, an instance of FileConfig should only exist once and be passed around by reference.

Reply:
@edwardalee Okay, thank you for catching that!

@cmnrd There is a quirk to the way .c files are generated for federated programs because multiple programs (federates) would need to be created from a single .lf file. The quirk is that each program will be called the same name as its variable (e.g., `a = new Count()` will be called federate a) and will have its own generated `.c` file. Therefore, each federate needs its own CMakeLists.txt. To avoid any conflicts, the generated files for each federate would need to be put in their own separate folder.

For example, for [DistributedCount.lf](https://github.com/icyphy/lingua-franca/blob/master/test/C/src/federated/DistributedCount.lf), the generated folder structure will look like:
```
DistributedCount % tree -L 2
.
├── c
│   ├── DistributedCount_c.c
│   ├── core
│   └── ctarget.h
└── p
    ├── DistributedCount_p.c
    ├── core
    └── ctarget.h

4 directories, 4 files
```

The normal FileConfig does not understand this structure, and assumes that the `src-gen` directory for DistibutedCount.lf is `DistributedCount`. This federated FileConfig will spring into action only if the program is federated and will change the `src-gen` directory to correctly reflect the federate's.